bin/gvs

```diff
@@ -95,30 +95,53 @@ for REPO in "${REPOSITORIES[@]}"; do
 
   # Checkout
   if $DO_CHECKOUT; then
-    # Delete composer dependencies from WooCommerce
-    if [ -f woocommerce.php ]; then
-      $VERBOSE_OUTPUT && echo "Delete composer packages from $(pwd)" || true
-      rm -f composer.lock && rm -rf vendor
-    fi
-
     # Checkout target
     git checkout --force --quiet "$REPO_TARGET"
 
-    # Install composer dependencies for WooCommerce
-    if [ -f woocommerce.php ]; then
-      ORIGINAL_VERSION=$(composer --version | cut -d " " -f 3)
-      if [ "${ORIGINAL_VERSION::1}" == "2" ]; then
-        echo "Switching composer version to install WooCommerce dependencies"
-        composer_self_update 1
-      fi
-      $VERBOSE_OUTPUT && echo "Install composer packages to $(pwd)" || true
-      composer install --no-suggest --no-dev --quiet
-      NEW_VERSION=$(composer --version | cut -d " " -f 3)
-      if [ "${ORIGINAL_VERSION::1}" != "${NEW_VERSION::1}" ]; then
-        echo "Switching composer version to install WooCommerce dependencies"
-        composer_self_update rollback
+    # Directories to look for a composer.json file in
+    declare -a dirs=("$REPO_PATH" "$REPO_PATH/plugins/woocommerce")
+    for dir in "${dirs[@]}"; do
+      if [ -f "$dir/composer.json" ]; then
+        $VERBOSE_OUTPUT && echo "composer.json exists at $dir/composer.json"
+        cd "$dir"
+
+        # Clean up files and then install composer dependencies
+
+        if [ -f "$dir/composer.lock" ] && [ $(git ls-files composer.lock | wc -l) -eq "0" ]; then
+          $VERBOSE_OUTPUT && echo "$dir/composer.lock is not committed to git. Deleting."
+          rm -f "$dir/composer.lock"
+        fi
```

Comment:
This logic repeated multiple times, please extract to a function

Also please put `$(...)` structures inside double qoute